packages/api/src/controllers/stripe.ts

```diff
@@ -0,0 +1,720 @@
+    streaming: product?.usage[1].limit,
+    storage: product?.usage[2].limit,
+  };
+
+  const overUsage = {
+    TotalUsageMins: Math.max(usage?.TotalUsageMins - limits.transcoding, 0),
+    DeliveryUsageMins: Math.max(usage?.DeliveryUsageMins - limits.streaming, 0),
+    StorageUsageMins: Math.max(usage?.StorageUsageMins - limits.storage, 0),
+  };
+
+  return overUsage;
+};
+
+const getBillingUsage = async (
+  userId,
+  fromTime,
+  toTime,
+  baseUrl,
+  adminToken
+) => {
+  // Fetch usage data from /data/usage endpoint
+  const usage = await fetch(
+    `${baseUrl}/api/data/usage/query?${qs.stringify({
+      from: fromTime,
+      to: toTime,
+      userId: userId,
+    })}`,
+    {
+      headers: {
+        Authorization: `Bearer ${adminToken}`,
+      },
+    }
+  ).then((res) => res.json());
+
+  return usage;
+};
+
+// Webhook handler for asynchronous events.
+app.post("/webhook", async (req, res) => {
+  const sig = req.headers["stripe-signature"];
+
+  let event;
+
+  try {
+    const secret = req.config.stripeWebhookSecret;
+    event = req.stripe.webhooks.constructEvent(req.body, sig, secret);
+  } catch (err) {
+    return res.status(400).send(`Webhook Error: ${err.message}`);
+  }
+
+  if (event.type === "invoice.created") {
+    let invoice = event.data.object;
+
+    if (invoice.status !== "draft") {
+      // we don't need to do anything
+      return res.sendStatus(200);
+    }
+
+    const [users] = await db.user.find(
+      { stripeCustomerId: invoice.customer },
+      { useReplica: false }
+    );
+
+    if (users.length < 1) {
+      res.status(404);
+      return res.json({ errors: ["user not found"] });
+    }
+
+    const user = users[0];
+
+    const usageRes = await db.stream.usage(
+      user.id,
+      invoice.period_start,
+      invoice.period_end,
+      {
+        useReplica: false,
+      }
+    );
+
+    // Invoice items based on usage
+    await Promise.all(
+      products[user.stripeProductId].usage.map(async (product) => {
+        if (product.name === "Transcoding") {
+          let quantity = Math.round(usageRes.sourceSegmentsDuration / 60);
+          await req.stripe.invoiceItems.create({
+            customer: user.stripeCustomerId,
+            invoice: invoice.id,
+            currency: "usd",
+            period: {
+              start: invoice.period_start,
+              end: invoice.period_end,
+            },
+            subscription: user.stripeCustomerSubscriptionId,
+            unit_amount_decimal: product.price * 100,
+            quantity,
+            description: product.description,
+          });
+        }
+      })
+    );
+  } else if (event.type === "invoice.payment_failed") {
+    // Notify via email
+    const invoice = event.data.object;
+    const [users] = await db.user.find(
+      { stripeCustomerId: invoice.customer },
+      { useReplica: false }
+    );
+
+    // notify help@livepeer.org
+    await sendgridEmailPaymentFailed({
+      email: "help@livepeer.org",
```

Comment:
Might be worth a CLI arg for this.

Also, I think the implementation isn't this an array?
```ts
  const [supportName, supportEmail] = email;
```